wgpu-core/src/device/mod.rs

```diff
@@ -329,6 +344,18 @@ pub enum DeviceError {
     DeviceMismatch(#[from] Box<DeviceMismatch>),
 }
 
+impl WebGpuError for DeviceError {
+    fn webgpu_error_type(&self) -> ErrorType {
+        match self {
+            DeviceError::DeviceMismatch(e) => e.webgpu_error_type(),
+            Self::Invalid(_) => ErrorType::Validation,
+            Self::ResourceCreationFailed => ErrorType::Internal { device_lost: false },
+            Self::Lost => ErrorType::Internal { device_lost: true },
```

Comment:
We should probably add a `Lost` variant to `ErrorType` that callers can just ignore since the lost callback has been called or will be called.